blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -132,8 +170,55 @@ Blockly.ContractEditor.prototype.create_ = function() {
           goog.bind(this.isBlockInFunctionArea, this),
           this.hiddenDefinitionBlocks_);
         this.position_();
+      }, this),
+      highlightBox: sharedHighlightBox,
+      placeContentCallback: goog.bind(function (currentY) {
+        if (this.flyout_) {
+          currentY += this.flyout_.getHeight();
+          this.flyout_.customYOffset = currentY;
+          this.flyout_.position_();
+        }
+
+        currentY += FUNCTION_BLOCK_VERTICAL_MARGIN;
+
+        if (this.functionDefinitionBlock) {
+          var fullWidth = Blockly.modalBlockSpace.getMetrics().viewWidth;
+          var functionDefinitionX = Blockly.RTL ? fullWidth - FRAME_MARGIN_SIDE : FRAME_MARGIN_SIDE;
+          this.functionDefinitionBlock.moveTo(
+            functionDefinitionX, currentY);
+          currentY += this.functionDefinitionBlock.getHeightWidth().height;
+        }
+
+        return currentY + FUNCTION_BLOCK_VERTICAL_MARGIN;
       }, this)
     });
+
+  this.allSections_.set(CONTRACT_SECTION_NAME, this.contractSectionView_);
+  this.allSections_.set(EXAMPLES_SECTION_NAME, this.examplesSectionView_);
+  this.allSections_.set(DEFINITION_SECTION_NAME, this.definitionSectionView_);
+};
+
+Blockly.ContractEditor.prototype.firstOpenLevelConfig_ = function (levelConfig) {
+  this.allSections_.forEach(function (sectionView, sectionName) {
+    var highlight = levelConfig[sectionName + HIGHLIGHT_CONFIG_SUFFIX];
+    if (highlight === TRUE_CONFIG_VALUE) {
+      this.setSectionHighlighted(sectionView);
+    }
+    var collapse = levelConfig[sectionName + COLLAPSE_CONFIG_SUFFIX];
+    sectionView.setCollapsed_(collapse === TRUE_CONFIG_VALUE);
+  }, this);
+};
+
+
+/** @override */
+Blockly.ContractEditor.prototype.openWithLevelConfiguration = function (levelConfig) {
+  this.levelConfigForFirstOpen_ = levelConfig;
+  Blockly.ContractEditor.superClass_.openWithLevelConfiguration.call(this, levelConfig);
+};
+
+Blockly.ContractEditor.prototype.setSectionHighlighted = function (viewToHighlight) {
+  this.allSections_.forEach(function (view) { view.setHighlighted(false) }, this);
```

Comment:
generally would prefer this to be on separate lines, but nbd. also, should have a semicolon after setHighlighted